subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/IgnoredPathCompareStrategy.java

```diff
@@ -87,7 +87,7 @@ protected boolean doVisitChangesSince(ChangeVisitor visitor, Map<String, FileSys
         Collections.sort(unaccountedForPreviousEntries, ENTRY_COMPARATOR);
         for (Map.Entry<HashCode, FilePathWithType> unaccountedForPreviousEntry : unaccountedForPreviousEntries) {
             FilePathWithType removedFile = unaccountedForPreviousEntry.getValue();
-            if (!visitor.visitChange(DefaultFileChange.removed(removedFile.getAbsolutePath(), propertyTitle, removedFile.getFileType()))) {
+            if (!visitor.visitChange(DefaultFileChange.removed(removedFile.getAbsolutePath(), propertyTitle, removedFile.getFileType(), IgnoredPathFingerprintingStrategy.IGNORED_PATH))) {
```

Comment:
This should be handled by path-less factory methods as well I think, though it's better to pass a constant than `""`.

Reply:
I am not convinced that a normalized path-less factory method is much better than passing in the constant. Is it OK for you if we use the constant for now?

Reply:
If we use the same constant everywhere, that's already better.